recipe-server/client/control_new/components/common/BaseAntForm.js

```diff
@@ -0,0 +1,94 @@
+import autobind from 'autobind-decorator';
+import { Form, message } from 'antd';
+import React, { PropTypes as pt } from 'react';
+
+
+/**
+ * Base class for antd-based forms that handles displaying errors
+ * from the API.
+ */
+export default class BaseAntForm extends React.Component {
+  static propTypes = {
+    form: pt.object.isRequired,
+    onSubmit: pt.func.isRequired,
+    errors: pt.object,
+  }
+
+  @autobind
+  handleSubmit(event) {
+    event.preventDefault();
+    this.triggerSubmit();
+  }
+
+  @autobind
+  async triggerSubmit() {
+    try {
+      const values = await this.validateFields();
+      this.props.onSubmit(values);
+    } catch (error) {
+      message.error('Could not validate form. Please correct the errors below.');
+    }
+  }
+
+  @autobind
+  validateFields() {
+    return new Promise((resolve, reject) => {
+      this.props.form.validateFields((error, values) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve(values);
+        }
+      });
+    });
+  }
+
+  /**
+   * Convenience wrapper for Form.Item and form.getFieldDecorator.
+   *
+   * Example:
+   *
+   * class MyForm extends BaseAntForm {
+   *   render() {
+   *     return (
+   *       <Form onSubmit={this.handleSubmit}>
+   *         <this.FormItem name="myfield" label="My Field" initialValue="foo">
+   *           <Input />
+   *         </this.FormItem>
+   *       </Form>
+   *     );
+   *   }
+   * }
+   */
+  @autobind
+  FormItem({ children, name, initialValue, rules, config = {}, ...customItemProps }) {
```

Comment:
I understand what you're doing here, but I think this is a _little_ weird. Since this is essentially its own component (takes children, props, etc), my instinct is to split this off into its own class/file instead of nesting it here, but I feel like there's a better solution than that. I'll think about it some more.